woocommerce/woocommerce
# CYS: Update Block Toolbar Position

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR updates the Block Toolbar Position: now it is rendered inside the pattern.

The current BlockPopover component doesn't natively support this positioning. To achieve this placement, we've implemented a custom solution for managing the popover's location.

| Before | After |
|--------|--------|
| ![image](https://github.com/woocommerce/woocommerce/assets/4463174/e27a07e2-8acb-4f4c-8a59-8031aacf96e1) | ![image](https://github.com/woocommerce/woocommerce/assets/4463174/d5bb9f33-3cbd-4bce-9b01-42245d261826) | 


Closes #48551.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that you have the last version of Gutenberg enabled.
2. Make sure you have the WooCommerce beta tester plugin installed.
3. Head over to Tools > WCA Test Helper > Features.
4. Make sure you see the `pattern-toolkit-full-composability` on the list.
5. Enable it.
6. Head over to WooCommerce > Home > Customize your store.
7. Click on Start designing.
8. Click on "Design your Homepage".
9. Click on a pattern.
10. Ensure that the tooltip is rendered as in the "after" column.
11. Scroll the page down ensure
12. Ensure that the tooltip is visible.
13. Switch Pattern.
14. Ensure that the tooltip position changes.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

CYS: Update Block Toolbar Position

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
